### ISSUE TITLE
Update Filterable MultiSelect to work in new format

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#7861]
The Filterable MultiSelect component currently has several issues that need to be addressed similarly to the work we need for Combobox for WCAG 2.1 AA requirements. In particular, this component should:

- [ ] Allow selection via keyboard (this is likely a regression)
- [ ] Allow closing via the ESC key: https://github.com/carbon-design-system/carbon/issues/7197#issuecomment-781001423

**Comment by mashenka123** [comment]
This bug is blocking our bug https://github.ibm.com/alchemy-containers/armada-ui/issues/5904

**Comment by mashenka123** [comment]
@joshblack This is still blocking Kubernetes and Registry accessibility compliance, and this is the last one, could you please fix this with priority? Thank you!

**Comment by joshblack** [comment]
Hi @mashenka123! 👋  This issue is up in a PR right now and should be merged in soon 😄
